### PR TITLE
openbugs: switch to new upstream

### PR DESCRIPTION
**Pull request by b-rodrigues**
## Description of changes

The download page for OpenBUGS changed, and so the build fails. This PR instead downloads the source from the archived Github mirror. I also listed 'x86_64-linux" as supported platforms. My understanding is that a 32-bits binary gets built, but is usable from a 64 bits linux as well.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by b-rodrigues** [comment]
Result of `nixpkgs-review pr 316714` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>openbugs</li>
  </ul>
</details>

**Review by andres-nav — APPROVED** [review]

**Comment by b-rodrigues** [comment]
Change since the last review: changed `platform.linux` to `platforms = [ "i686-linux" "x86_64-linux"];` because `platform.linux` also includes "aarch64-linux" and this package cannot be installed on arm.

File: pkgs/applications/science/machine-learning/openbugs/default.nix [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
+    owner = "jsta";
+    repo = "openbugs";
+    rev = "cd921342ba13ee89ee60f9aebd2e96c42bd59ae3";
+    sha256 = "sha256-11LrScN1kvtq0Fo7RWGjbQO0U5b5brCbipl5pdZnrFs=";
   };
 
   meta = with lib; {
-    description = "Open source program for Bayesian modelling based on MCMC";
+    description = "OpenBUGS is a software package for performing Bayesian analysis and
+simulation using Markov Chain Monte Carlo.";
```
Comment (FliegendeWurst) [inline pkgs/applications/science/machine-learning/openbugs/default.nix:21]:
```suggestion
    description = "Software package for performing Bayesian analysis and simulation using Markov Chain Monte Carlo";
```

File: pkgs/applications/science/machine-learning/openbugs/default.nix [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
     homepage = "https://www.mrc-bsu.cam.ac.uk/software/bugs/openbugs/";
-    maintainers = with maintainers; [ andresnav ];
+    changelog = "https://github.com/jsta/openbugs/blob/master/ChangeLog";
+    platforms = [ "i686-linux" "x86_64-linux"];
```
Comment (FliegendeWurst) [inline pkgs/applications/science/machine-learning/openbugs/default.nix:24]:
```suggestion
    platforms = [ "i686-linux" "x86_64-linux" ];
```

**Review by andres-nav — APPROVED** [review]
Thank you for the pull request. All good!

File: pkgs/applications/science/machine-learning/openbugs/default.nix (hunk 1 above)
Comment (b-rodrigues) [reply pkgs/applications/science/machine-learning/openbugs/default.nix]: thanks!

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**